src/traces/scatter/defaults.js

```diff
@@ -63,6 +66,11 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         if(!subTypes.hasLines(traceOut)) handleLineShapeDefaults(traceIn, traceOut, coerce);
     }
 
+    if(traceOut.fill === 'tonext' || traceOut.fill === 'toself') {
+        dfltHoverOn.push('fills');
```

Comment:
Ternary is easier here because it doesn't even *have* a `hovermode: 'x'`... But I'd also argue that if you have a scatter trace with fill tonext or toself, you're making the statement "I'm enclosing an area" which doesn't make much sense in conjunction with comparing different traces at the same x value. For one, your outline trace is *itself* generally going to have two (or more) points at the same x value (or 2 crossings of the same x value anyway, on opposite sides of the region) which is a situation that has always confused people ("why do I never see hover data for the second point?")

Not sure if there's an issue about it, but we've talked in the past about using trace info to make a smarter default `hovermode` choice, and I think this makes that case even stronger. `hovermode: 'x'` should only really be the default when you have multiple traces all with monotonically increasing (or decreasing) x (or position, like horizontal bars) data. I don't think that would be too intrusive, `hovermode` is a layout property but it is all about exposing the trace data anyway. If we did that, this concern would take care of itself.

I suppose in the meantime we could put `hoveron` back to the way it was before as a default: only have `hoveron: 'fills'` if you have an area fill *and* no explicit points, otherwise `hoveron: 'points'`, never default to both. But I still think I'd prefer to leave it as is now, because it matches the meaning implied by your trace attributes.